src/core/socket.c

```diff
@@ -898,30 +901,35 @@ static int instance_from_socket(int fd, unsigned nr, char **instance) {
 
         case AF_UNIX: {
                 struct ucred ucred;
-                int k;
 
-                k = getpeercred(fd, &ucred);
-                if (k >= 0) {
-                        if (asprintf(&r,
-                                     "%u-"PID_FMT"-"UID_FMT,
-                                     nr, ucred.pid, ucred.uid) < 0)
+                r = getpeercred(fd, &ucred);
```

Comment:
once #35839 is merged this could probably be written in a shorter way by using getpeerpidfd() and pidref_acquire_pidfd_id()

Reply:
i take this back, it doesn't get shorter. i tried and failed.